lib/midcom/helper/datamanager2/widget/tinymce.php

```diff
@@ -258,9 +260,32 @@ private function _get_tiny_configuration ()
 theme : "modern",
 menubar: false,
 statusbar: false,
-plugins : ["table save contextmenu link fullscreen image"],
+plugins : ["table save contextmenu link fullscreen image imagetools"],
 toolbar: "bold italic | bullist | link image | code fullscreen",
 extended_valid_elements : "a[name|href|target|title|onclick],img[class|src|border=0|alt|title|hspace|vspace|width|height|align|onmouseover|onmouseout|name|style],hr[class|width|size|noshade],font[face|size|color|style],span[class|align|style]",
+{$this->_get_imagetools_configuration()}
+        
+EOT;
+    }
+
+    /**
+     * Returns the imagetools configuration
+     */
+    private function _get_imagetools_configuration()
+    {
+        $hostname = midcom::get()->get_host_name();
+        $guid = null;
+        if(!empty($this->_type->storage->object))
+        {
+            $guid = $this->_type->storage->object->guid;
+        }
+    
```

Comment:
so what happens if `$guid` is `null`? did you test if upload works in this case?